src/coreclr/src/vm/threadsuspend.cpp

```diff
@@ -218,32 +218,30 @@ Thread::SuspendThreadResult Thread::SuspendThread(BOOL fOneTryOnly, DWORD *pdwSu
 
     while (TRUE) {
         StateHolder<MyEnterLogLock, MyLeaveLogLock> LogLockHolder(FALSE);
-
         CounterHolder handleHolder(&m_dwThreadHandleBeingUsed);
 
-        // Whether or not "goto retry" should YieldProcessor and __SwitchToThread
-        BOOL doSwitchToThread = TRUE;
-
         hThread = GetThreadHandle();
         if (hThread == INVALID_HANDLE_VALUE) {
             str = STR_UnstartedOrDead;
             break;
         }
-        else if (hThread == SWITCHOUT_HANDLE_VALUE) {
```

Comment:
Looking at the entire codebase, we still have at least one path which still sets the thread handle to SWITCHOUT_HANDLE_VALUE.  (Thread::DetachThread) We should either get rid of the SWITCHOUT_HANDLE_VALUE, or do something sensible.

Reply:
The uses are in debugger related code. I was not sure if that needs to handle fibers (as in - "if user uses fibers"). Now I am firmly sure that we do not promise any particular behavior if user uses fibers in the runtime, so we can't have a reasonable behavior when debugger encounters them.

I will go and remove all the usages. It is not something that we can handle anyways.

---
In reply to: [508046926](https://github.com/dotnet/runtime/pull/34666#discussion_r508046926) [](ancestors = 508046926)

Reply:
Actually, no. Now I remember why I left remaining uses of SWITCHOUT_HANDLE_VALUE. In all cases that remain it is treated as an invalid state. - similar to INVALID_HANDLE_VALUE. We don't really "handle" fiber's in these cases.

I think it is better that way as it will lead to a more deterministic/early failure if fibers are used. 
As opposed - if we treat it as a valid thread handle we may run more code and eventually fail in some way anyways, but failing earlier seems better. 


---
In reply to: [508123837](https://github.com/dotnet/runtime/pull/34666#discussion_r508123837) [](ancestors = 508123837,508046926)

Reply:
That is my exact concern. In this case SWITCHOUT_HANDLE_VALUE isn't treaded like INVALID_HANDLE_VALUE. One way of fixing it would be to change the if check above so it checks for SWITCHOUT_HANDLE_VALUE as well as INVALID_HANDLE_VALUE.

Reply:
Ah, yes. That we can do. Where we check for INVALID_HANDLE_VALUE and used to check for SWITCHOUT_HANDLE_VALUE, treating both the same now does make sense. I will check other places too. Thanks!

Reply:
For the `Thread::DetachThread` - I think we should use INVALID_HANDLE_VALUE. I will check if there are any issues with that.

Reply:
I have removed the SWITCHOUT_HANDLE_VALUE entirely. Without explicit fiber support it is equivalent to INVALID_HANDLE_VALUE and only causes confusion.

---
In reply to: [509813214](https://github.com/dotnet/runtime/pull/34666#discussion_r509813214) [](ancestors = 509813214)